### PR TITLE
Refactor fence to wall

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Will keep working on this as its still a bit of a mess.

### CODE REVIEW

File: src/openrct2/world/map_animation.c [hunk 1]
```diff
@@ -456,7 +456,7 @@ static bool map_animation_invalidate_wall_door(sint32 x, sint32 y, sint32 baseZ)
 	if (gCurrentTicks & 1)
 		return false;
 
-	bool wasInvalidated = false;
+	bool removeAnimation = true;
```
Comment (wolfreak99) [inline src/openrct2/world/map_animation.c]: Does this detect if it's an animation or does it just play through the animation? If it's the latter, wouldn't it be more appropriate to have it "finishAnimation"? Remove sounds like it makes it static forever. Just my opinion though, what do you think?

File: src/openrct2/paint/map_element/fence.c [hunk 2]
```diff
@@ -182,7 +182,9 @@ void fence_paint(uint8 direction, sint32 height, rct_map_element * map_element)
     if (sceneryEntry->wall.flags & WALL_SCENERY_IS_DOOR) {
         rct_xyz16 offset;
         rct_xyz16 boundsR1, boundsR1_, boundsR2, boundsR2_, boundsL1, boundsL1_;
-        uint8 animationFrame = (map_element->properties.fence.item[2] >> 3) & 0x1F;
+		uint8 animationFrame = fence_get_animation_frame(map_element);
+		// Add the direction as well
+		animationFrame |= (map_element->properties.fence.item[2] & 0x80) >> 3;
```
Comment (wolfreak99) [inline src/openrct2/paint/map_element/fence.c]: Could you fix the alignment in these 3 lines?

File: src/openrct2/world/map_animation.c (hunk 1 above)
Comment (duncanspumpkin) [inline src/openrct2/world/map_animation.c]: All of the animation functions either remove the animation entry at the end or keep it for the next frame to decide. Finish animation would not quite make sense in this situation.

**Comment by LRFLEW** [comment]
Git wasn't letting me push, so I did it the old fashioned way and made a PR. @duncanspumpkin Merge duncanspumpkin/OpenRCT2#13 and remove the label please 😄.

File: src/openrct2/paint/map_element/fence.c [hunk 3]
```diff
@@ -182,7 +182,9 @@ void fence_paint(uint8 direction, sint32 height, rct_map_element * map_element)
     if (sceneryEntry->wall.flags & WALL_SCENERY_IS_DOOR) {
         rct_xyz16 offset;
         rct_xyz16 boundsR1, boundsR1_, boundsR2, boundsR2_, boundsL1, boundsL1_;
-        uint8 animationFrame = (map_element->properties.fence.item[2] >> 3) & 0x1F;
+		uint8 animationFrame = wall_element_get_animation_frame(map_element);
```
Comment (janisozaur) [inline src/openrct2/paint/map_element/fence.c]: Wrong indent

**Pull request merged**